src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_CAPACITY = 10;
+    private int arraySize;
+    private Object[] values;
+
+    public ArrayList() {
+        arraySize = 0;
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void add(T value) {
-
+        if (isNotEnoughSpace()) {
+            resize();
+        }
+        values[arraySize] = value;
+        arraySize++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > arraySize || index < 0) {
+            throw new ArrayIndexOutOfBoundsException("wrong index");
+        }
+        if (isNotEnoughSpace()) {
+            resize();
+        }
+        System.arraycopy(values, index, values, index + 1, arraySize - index);
+        values[index] = value;
+        arraySize++;
     }
 
     @Override
     public void addAll(List<T> list) {
 
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (!(index >= arraySize || index < 0)) {
```

Comment:
you are making the same check in get(), put() and remove()
let's move it to a separate method